template/.gitlab-ci.yml

```diff
@@ -1,46 +1,76 @@
 stages:
 - Install dependencies
 - Code Quality
+- Code Quality Report
+
+# Global workflow config
+workflow:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "push"'
 
 variables:
   # Needed for NPM Cache
   npm_config_cache: "$CI_PROJECT_DIR/.npm"
+  CYPRESS_CACHE_FOLDER: "$CI_PROJECT_DIR/.cache/Cypress"
 
 ########################################################################
 #  BEGIN CACHE                                                         #
 ########################################################################
 
 .NPM Cache: &npm_cache
-  key: npm-cache
+  key: "npm-cache-$CI_COMMIT_REF_SLUG"
   paths:
     - .npm
     - node_modules
   policy: pull-push
 
 .Composer Cache: &composer_cache
-  key: composer-cache
+  key: "composer-cache-$CI_COMMIT_REF_SLUG"
   paths:
     - composer.phar
     - vendor
     - web/wp
-    - web/wp-content/mu-plugins/wp-mu-cleanup
-     <%_ if (wpRocket) { _%>
-    - web/wp-content/plugins/wp-rocket
-    <%_ } _%>
-    <%_ if (wordfence) { _%>
-    - web/wp-content/plugins/wordfence
-    <%_ } _%>
-    <%_ if (classicEditor) { _%>
-    - web/wp-content/plugins/classic-editor
-    <%_ } _%>
-    <%_ if (yoastSeo) { _%>
-    - web/wp-content/plugins/wordpress-seo
-    <%_ } _%>
-    <%_ if (acf) { _%>
-    - web/wp-content/plugins/advanced-custom-fields-pro
-    <%_ } _%>
   policy: pull-push
 
+.NPM default config: &npm_default_config
+  image: node:12
+  cache:
+    <<: *npm_cache
+    policy: pull
+  rules:
+    - changes:
+        - package.json
+        - package-lock.json
+        - meta.config.js
+        - .babelrc.js
+        - .eslintrc.js
+        - .postcssrc.js
+        - .prettierrc.js
+        - .stylelintrc.js
+        - tailwind.config.js
+        - .gitlab-ci.yml
+        - web/wp-content/themes/**/*.{js,scss,vue}
+
+.Composer default config: &composer_default_config
+  image: php:7.2
+  before_script:
+    - apt-get update
+    - apt-get install -y git wget unzip zip
+    - wget https://raw.githubusercontent.com/composer/getcomposer.org/6e874537c7ff87f03f99dc8dea1f2c48e4274da3/web/installer -O - -q | php -- --quiet
```

Comment:
We need to add the version for the Composer installation, we are not ready for Composer 2.

```bash
wget https://raw.githubusercontent.com/composer/getcomposer.org/6e874537c7ff87f03f99dc8dea1f2c48e4274da3/web/installer -O - -q | php -- --quiet --version=1.10.16
```